node/src/components/consensus/protocols/highway/round_success_meter.rs

```diff
@@ -160,6 +139,19 @@ impl<C: Context> RoundSuccessMeter<C> {
         new_exp
     }
 
+    /// Returns an instance of `Self` for the new era: resetting the counters where appropriate.
+    pub fn next_era(&self, era_start_timestamp: Timestamp) -> Self {
+        let current_round_id = round_id(era_start_timestamp, self.current_round_exp).millis();
+        Self {
+            rounds: Default::default(),
+            current_round_id,
+            proposals: Default::default(),
```

Comment:
It makes sense to reset the proposals here, but not the rounds - in fact, continuity of the `rounds` collection is the essence of this issue. But the good news is that changing this to `rounds: self.rounds.clone()` should be the only thing needed to make this work ;)

Reply:
Of course. Thanks :)